### PR TITLE
Fix clip and eas conditionals

### PR DESCRIPTION
**Pull request by Forst**

### CODE REVIEW

**Comment by EliasOenal** [comment]
Thanks for the pull request. Given that I'm not too familiar with these two decoders, could you please give me a short explanation of what was fixed and where the new magic numbers came from?

**Comment by Forst** [comment]
I'm not familiar with these decoders either, but I was trying to follow the logic of what the original author tried to do.

Both `eas` and `clip` modules raised warnings during compilation about conditions that always evaluate to the same value (either true or false), this should normally never happen:

```clang
clip.c:345:26: warning: bitwise comparison always evaluates to false [-Wtautological-bitwise-compare]
                                else if ( (i & 0x70) == 7) {
                                          ~~~~~~~~~~~^~~~
clip.c:306:24: warning: bitwise comparison always evaluates to false [-Wtautological-bitwise-compare]
                                else if ( (i & 60) == 2 ) {
                                          ~~~~~~~~~^~~~
clip.c:303:24: warning: bitwise comparison always evaluates to false [-Wtautological-bitwise-compare]
                                else if ( (i & 60) == 1 ) {
                                          ~~~~~~~~~^~~~
clip.c:278:26: warning: bitwise comparison always evaluates to false [-Wtautological-bitwise-compare]
                                else if ( (i & 0x60) == 2 ) {
                                          ~~~~~~~~~~~^~~~
clip.c:275:26: warning: bitwise comparison always evaluates to false [-Wtautological-bitwise-compare]
                                else if ( (i & 0x60) == 1 ) {
                                          ~~~~~~~~~~~^~~~
demod_eas.c:125:19: warning: overlapping comparisons always evaluate to true [-Wtautological-overlap-compare]
   if (data >= 32 || data <= 126)
       ~~~~~~~~~~~^~~~~~~~~~~~~~
```

In `eas`, the condition I fixed checks whether the character is printable. Since we're checking a range, multiple checks are ANDed, not ORed. This has already been mentioned [here](https://github.com/EliasOenal/multimon-ng/issues/99#issuecomment-379806141).

In `clip`, the conditions look like this:

* `(i & 0x60) == 1`
* `(i & 0x60) == 2`
* `(i & 60) == 0`
* `(i & 60) == 1`
* `(i & 60) == 2`
* `(i & 0x70) == 7`

All of those would always evaluate to false, because the author never accounted for bit shifts.

Let's take the first conditional as an example

<table>
<tr><th>0x60</th><td>0</td><td>1</td><td>1</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td></tr>
<tr><th>1</th><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td><td>1</td></tr>
</table>

Obviously, this will always evaluate to 0. Instead, what I think they wanted to check is the following:

<table>
<tr><th>0x60</th><td>0</td><td>1</td><td>1</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td></tr>
<tr><th>0x20</th><td>0</td><td>0</td><td>1</td><td>0</td><td>0</td><td>0</td><td>0</td><td>0</td></tr>
</table>

Basically, they forgot to account for the bit shift. In some cases they also used decimal `60` instead of `0x60`, which is most likely incorrect.

**Comment by EliasOenal** [comment]
Thanks a lot, especially for the thorough explanation.
